### PR TITLE
Arcade machine class

### PR DESCRIPTION
**Pull request by studioant**
Hey team, 
I've integrated the games and options menus and refactored some class functions to handle the changes.
All high-level functionality is now handled in the ArcadeMachine class instead of Program.cpp.
Main menu is now being drawn using Grid.
Options menu is selectable.
ESC can't be used to exit main menu, must use Exit button.

### CODE REVIEW

File: gridlayout.h [hunk 1]
```diff
@@ -196,6 +198,10 @@ class Grid
                         }
                         sprite_set_x(_grid[index].button->btn(), x);
                         sprite_set_y(_grid[index].button->btn(), y);
+                        // Update Button position of button
+                        this->_grid[index].button->set_x(sprite_x(this->_grid[index].button->btn()));
+                        this->_grid[index].button->set_y(sprite_y(this->_grid[index].button->btn()));
```
Comment (Delcari) [inline gridlayout.h:203]: Unsure why the position of the sprite isn't just acquired from use of the [sprite_position](https://splashkit.io/api/sprites/#sprite-position) function, to ensure consistency is maintained across the code base.

File: gridlayout.h [hunk 2]
```diff
@@ -42,6 +42,8 @@ class Grid
 
 public:
     // Constructor:
+    // Default Constructor
+    Grid(){}
```
Comment (Delcari) [inline gridlayout.h:46]: This constructor may cause issues as most of the properties are private and a handful cannot be modified outside of the class

**Review by Delcari — APPROVED** [review]
Changes look good, nice to see some documentation
Please review aforementioned comments, and merge when ready!

**NOTE:** I have fixed the issues with gridlayout background and pushed changes.

**Review by sarahgos — APPROVED** [review]
Happy with the high-level functionality being moved to Arcade Machine class, looks like it's all nicely structured Anthony. As far as I can see it's looking good and I'm happy for it to be merged. Good job mate!

File: gridlayout.h (hunk 1 above)
Comment (studioant) [reply gridlayout.h:203]: _set_x/set_y_ was used to maintain consistency with existing _sprite_set_x/sprite_set_y_ used above.

File: gridlayout.h (hunk 2 above)
Comment (studioant) [reply gridlayout.h:46]: Default constructor is required to allow the object to be a class property.

**Pull request merged**